### ISSUE TITLE
Can't copy file unless ACL persmission is given

### ISSUE THREAD

**Issue by igorsantos07** [thephpleague/flysystem-aws-s3-v3#170]
This seems(?) related to #128...

I'm trying to copy an image in my S3 bucket, but I'm getting 403 Forbidden. After some digging, I understood my user wasn't given ACL permissions, which seems to be required to keep the same permissions the original file had.

I guess there should be an option to force a permission level in those cases, or only try to fetch if a default is not given? The original AWS SDK defaults to `private` if the permission is not given during copy operations... Not to mention that, if I want all my files to be considered `private`, there's no need for an extra API request to state the obvious.

**Comment by frankdejonge** [comment]
Couldn't you just set the default ACL options in the contructor config?

**Comment by igorsantos07** [comment]
I thought about that, but it doesn't seem it would make a difference: [copy() goes directly into getRawVisibility()](https://github.com/thephpleague/flysystem-aws-s3-v3/blob/master/src/AwsS3Adapter.php#L413). Or did I miss something?

**Comment by frankdejonge** [comment]
So to understand this better. You're getting a 403 when fetching the visibility of an object you're trying to copy? But you're able to read the object or copy without visibility? Could be that you need additional permission in AWS S3 client.

**Comment by igorsantos07** [comment]
Yeah, it's probably a lack of permission to read ACL details, but that
shouldn't be mandatory for this. There should be an option to always use
the default visibility, instead of asking for the current one during the
copy operation.

**Comment by rvanlaak** [comment]
![image](https://user-images.githubusercontent.com/2707563/163406570-95010f86-51a3-40a4-9a6d-d32223d252ae.png)

Given the above, Amazon seems to recommend disabling ACL. Can we prevent the Flysystem adapter or the S3Client from calling ACL related code? And thereby not having to modify the bucket settings?

The [old adapter docs](https://flysystem.thephpleague.com/v1/docs/adapter/aws-s3-v3/) seem to mention ACL, while [the new adapter docs](https://flysystem.thephpleague.com/docs/adapter/aws-s3-v3/) do not anymore.

**Comment by AYakimuk** [comment]
Hi to all,

I had similar problem. I couldn't copy object in S3 storage because of ACL disabled.
[There is](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ensure-object-ownership.html#ensure-object-ownership-bucket-policy) you can read about using PUT operations when  ACL disabled.

This helped me in my case:

`new AwsS3V3Adapter(
            ...
            options: [
                'ACL' => 'bucket-owner-full-control'
            ]
        )`

I hope it will useful.

**Comment by rvanlaak** [comment]
@AYakimuk So if I understand correctly, adding that constructor argument will disable the ACL checks?

**Comment by Smef** [comment]
You can set `'visibility' => 'noPredefinedVisibility',` and it looks like it won't try to write ACL, meaning you can leave the ACL disabled and the bucket ownership at the recommended setting.

**Comment by JshGrn** [comment]
> You can set `'visibility' => 'noPredefinedVisibility',` and it looks like it won't try to write ACL, meaning you can leave the ACL disabled and the bucket ownership at the recommended setting.

I am still getting ' The bucket does not allow ACLs', and do not want to enable ACL on the bucket, do you have this working?

**Comment by Smef** [comment]
Yes, adding that setting resolved this for me.

```
       s3 => [
            'driver' => env('ASSETS_STORAGE_DRIVER', 's3'),
            'key' => env('AWS_ACCESS_KEY_ID'),
            'secret' => env('AWS_SECRET_ACCESS_KEY'),
            'region' => env('AWS_DEFAULT_REGION'),
            'bucket' => env('AWS_BUCKET'),
            'url' => env('AWS_URL'),
            'endpoint' => env('AWS_ENDPOINT'),
            'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
            'throw' => false,
            'visibility' => 'noPredefinedVisibility',
        ]
```

**Comment by vinay-laerdalblr** [comment]
> Hi to all,
> 
> I had similar problem. I couldn't copy object in S3 storage because of ACL disabled. [There is](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ensure-object-ownership.html#ensure-object-ownership-bucket-policy) you can read about using PUT operations when ACL disabled.
> 
> This helped me in my case:
> 
> `new AwsS3V3Adapter( ... options: [ 'ACL' => 'bucket-owner-full-control' ] )`
> 
> I hope it will useful.

This configuration is very important for cross aws account(s). It should be highlighted enough in the document.

**Comment by raczekm** [comment]
In my opinion, the library needs work in this area.

Workaround for a non-public bucket is to add node `options` to the configuration in `filesystems.php` without `visibility` option. 

Working example:

```
        's3' => [
            'driver' => 's3',
            'key' => ... , 
            'secret' => ... , 
            'region' => ... , 
            'bucket' => ... , 
            'options' => [ 'ACL' => 'bucket-owner-full-control' ],
        ],
```

**Comment by tb-b** [comment]
Cloudflare R2 also doesn't support ACL so this doesn't work when in use with something like Statamic. 
` Error executing "PutObject". Header 'x-amz-acl' with value '' not implemented`
There need to be an option to completely disable x-amz-acl header
